.functional_ci_setup.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
```

Comment:
Can we move this file in some folder for the jenkins related stuff? Probablhy we will need more files in future, so will be good have them in one place.  Is there specific reason to have this file as hidden prefixed with dot?

Reply:
Well, this file isn't just for Jenkins - so I'm not sure what I'd call the subdirectory to put this in. It also doesn't have to be a hidden file - I just didn't want people to be tempted to execute it manually, since it won't do anything useful.

Reply:
Ok, lets create some specific folder for those jenkins scripts called .jenkins , .jenkins-ci or whatever better in your mind for example ?

Reply:
Yeah, again, this won't just be useful for Jenkins. And if we choose to use GitLab CI in the future, its configuration file has to be in the project's root directory - similar to TravisCI.

As it stands right now, there are two files that we could put in this directory. Could we maybe decide how to organize things if and when we have more?

Reply:
I am not against config files have in root directory if some system counts on them to be in root directory. But this file is not any configuration file for any CI system and it's used directly in pipeline (So I think it can be easily moved to some specific folder like ci-scripts if we don't like to use jenkins in the name). So as @clacroix12  already tried to do some restructure/cleanup of the project and cleaned everything to specific folders I don't like to have any extra python file in root of the project.  @clacroix12 thoughts?   I won't block this PR cause of this but still think we don't need it in the root.

Reply:
@petr-balogh The only thing I can think of is something like `.pipelines/` which would keep it arbitrary as far as what CI system is consuming it. A `.` dir to perhaps keep it out of the way and like @zmc said to keep people from trying to execute it manually.

Overall, I don't think it's a big deal to leave it as is but if we did want to have it in a directory that's how I would approach it.